### ISSUE TITLE
With GNU C++, line 85 of hpx/config/version.hpp causes link failure when linking application

### ISSUE THREAD

**Issue by dmarce1** [STEllAR-GROUP/hpx#2079]
When I build HPX with GNU C++ and then try to build an application with it, I get unresolved symbol errors coming from line 85 of hpx/config/version.hpp. Everything works when I comment this line out.

**Comment by hkaiser** [comment]
I assume (you have not shown the error) this linker error happens whenever HPX was compiled against a different version of Boost than you're trying to use in your application.

**Comment by dmarce1** [comment]
Here is the error, from trying to compile a hello world program (same error with octotiger):

CMakeFiles/octotiger_exe.dir/src/main.cpp.o: In function `(anonymous namespace)::check_hpx_version()':
/work/dmarce1/debug/include/hpx/config/version.hpp:85: undefined reference to 
`hpx::hpx_check_boost_version_105700'
collect2: error: ld returned 1 exit status


I looked at CMakeCache.txt in the hpx build directory, and it appears to have the correct boost. Its the only boost I've got on the system right now, except the default boost (1.41) which I've never been able to compile against. I will redo the compilation from scratch to make sure.
